cli/commands/base.ts

```diff
@@ -0,0 +1,26 @@
+export type OutputFormat = undefined | 'json';
+
+export abstract class BaseCommand {
```

Comment:
I have to admin that I'm not a big fan of classes and that I prefer typed structures (interface). But regardless, it's a style choice, I would love to understand more the choices made here:

- Why is this a class? 
- What if a command is just a function with a given interface and "reportProgress", "reportError" two helper function in a "lib/logger" or something like that?

Mostly curious to understand the vision here.

Reply:
> Why is this a class?

There are no hard blockers pushing us in the class direction. It does seemingly make it easier to keep track of `outputFormat` state and to enforce a structure for reporting statuses.

We could probably do this without too much hassle with functions, too. I'll give that a try tomorrow and see where we end up.